sql/src/main/java/io/crate/protocols/postgres/AbstractPortal.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres;
+
+import io.crate.action.sql.RowReceiverToResultReceiver;
+import io.crate.analyze.Analysis;
+import io.crate.analyze.Analyzer;
+import io.crate.exceptions.ReadOnlyException;
+import io.crate.executor.Executor;
+import io.crate.executor.transport.kill.TransportKillJobsNodeAction;
+import io.crate.operation.projectors.ResumeHandle;
+
+import java.util.UUID;
+
+public abstract class AbstractPortal implements Portal {
+
+    protected SessionData sessionData;
+    protected RowReceiverToResultReceiver rowReceiver = null;
+    protected final String name;
+
+    public AbstractPortal(String name,
+                          UUID jobId,
+                          String defaultSchema,
+                          Analyzer analyzer,
+                          Executor executor,
+                          TransportKillJobsNodeAction transportKillJobsNodeAction,
+                          boolean isReadOnly) {
+        this.name = name;
+        sessionData = new SessionData(jobId, defaultSchema, analyzer, executor,
+                                      transportKillJobsNodeAction, isReadOnly);
+    }
+
+    public AbstractPortal(String name, SessionData sessionData) {
+        this.name = name;
+        this.sessionData = sessionData;
+    }
+
+    abstract protected Type type();
```

Comment:
this is unused except of the `toString`. Could probably be a required argument in the constructor and then the type property can be removed.

Or since `toString` is mostly for debugging it would also be fine to just use `getClass().getSimpleName()`